yjit/src/invariants.rs

```diff
@@ -568,10 +569,12 @@ pub extern "C" fn rb_yjit_tracing_invalidate_all() {
             asm.jmp(patch.outlined_target_pos.into());
 
             cb.set_write_ptr(patch.inline_patch_pos);
+            cb.set_dropped_bytes(false);
```

Comment:
We seem to have a couple of places doing the `get_write_pos`/`has_dropped_bytes` and `set_pos`/`set_dropped_bytes` pattern. For safety and hopefully cleaner code, do you think it makes sense to provide a helper function like this:

```rust
cb.with_patch_pos(patch.inline_patch_pos, |cb| {
    asm.compile(cb);
    last_patch_end = cb.get_write_ptr().raw_ptr();
});
```

where `with_patch_pos` takes care of temporarily updating `write_pos`/`dropped_bytes` and restoring it?

Reply:
Since we have a case in `regenerate_branch` where we only sometimes restore and this place doesn't exactly fit that pattern (because of the loop) I don't like it that much.

Reply:
I think I actually thought about the same thing when touching that place the last time, so that makes sense.